ui/redux/actions/claims.js

```diff
@@ -152,31 +157,56 @@ export function doResolveUris(
  *
  * @param claimIds
  */
-export function doResolveClaimIds(claimIds: Array<string>) {
-  return (dispatch: Dispatch, getState: GetState) => {
+export function doResolveClaimIds(claimIds: Array<string>, returnCachedClaims?: boolean = true) {
+  return async (dispatch: Dispatch, getState: GetState) => {
     const state = getState();
-    const resolvedIds = Object.keys(selectById(state));
-    const idsToResolve = claimIds.filter((x) => !resolvedIds.includes(x));
+
+    const resolvingIds = new Set(selectResolvingIds(state));
+    const claimsById = selectClaimsById(state);
+    const resolvedClaims = {};
+
+    const idsToResolve = claimIds.filter((id) => {
+      if (resolvingIds.has(id)) {
+        return false;
+      }
+
+      if (returnCachedClaims) {
+        const claim = claimsById[id];
+
+        if (claim) {
+          resolvedClaims[claim.permanent_url] = claim;
+          return false;
+        }
+      }
+
+      return true;
+    });
 
     if (idsToResolve.length === 0) {
-      return Promise.resolve();
+      return Promise.resolve(resolvedClaims);
     }
 
-    return dispatch(
+    const isAuthenticated = selectUserVerifiedEmail(state);
+
+    const response = await dispatch(
       doClaimSearch(
         {
           claim_ids: idsToResolve,
           page: 1,
           page_size: Math.min(idsToResolve.length, 50),
+          include_is_my_output: isAuthenticated,
           no_totals: true,
         },
```

Comment:
:small_red_triangle:  In `doResolveClaimIds` via `claim_search`, not sure if always adding `include_is_my_output` when authenticated is the right things to do here, caching wise.

We previously removed that flag unless necessary.

_(cc: @tzarebczan fyi, in case I was thinking of something else, or things have changed)_